src/material/datepicker/calendar-body.ts

```diff
@@ -159,18 +165,31 @@ export class MatCalendarBody implements OnChanges, OnDestroy, AfterViewChecked {
   /** Width of an individual cell. */
   _cellWidth: string;
 
+  private _didDragSinceMouseDown = false;
+
```

Comment:
When testing it out with a mouse, I noticed that the preview line blinks when moving my cursor over to other date cells. This happens both with the old click pattern and when dragging and it isn't present on the `main` branch:

https://user-images.githubusercontent.com/4450522/197736211-cb241df0-fd56-4b52-adb9-c3c7541eb4c4.mov

Reply:
Hmm. I assume this is because of the mouseleave handlers. It wasn't actually clear to me why those were there since we'd pick up the next mouseenter right after. Any idea why I can't remove those?

Reply:
I see the slight behavior difference, but I do wonder why it was not happening before given this unchanged code in calendar body's _leaveHandler:

if (event.target && this._getCellFromElement(event.target as HTMLElement)) {
        this._ngZone.run(() => this.previewChange.emit({value: null, event}));
      }

Reply:
Actually, figured it out. We're detecting more parts of a cell than before. Have a workaround in place.